### PR TITLE
feat(sdk): enhance timeout logic and add timeout policies for init, resume and finish

### PR DESCRIPTION
**Pull request by kptkin**
Fixes WB-NNNN
Fixes #NNNN

Description
-----------
What does the PR do? This:

image

Testing
-------
How was this PR tested? Thoroughly.

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: tests/unit_tests/test_wandb_run.py [hunk 1]
```diff
@@ -454,3 +454,24 @@ def test_settings_unexpected_args_telemetry(runner, relay_server, capsys, user):
...
             assert 2 in telemetry.get("11", [])
             run.finish()
+
+
+@pytest.mark.parametrize(
+    "init_timeout",
+    [0.01, 3],
+)
+def test_init_timeout_allow_policy(wandb_init, capsys, init_timeout):
+    run = wandb.init(
```
Comment (kptkin) [inline tests/unit_tests/test_wandb_run.py:464]:
```suggestion
    run = wandb_init(
```

Comment (dmitryduev) [reply tests/unit_tests/test_wandb_run.py:464]: 🤦🏻‍♂️

File: wandb/sdk/internal/settings_static.py [hunk 2]
```diff
@@ -51,6 +51,8 @@ class SettingsStatic:
     host: Optional[str]
     username: Optional[str]
     _executable: str
+    run_url: Optional[str]
+    run_name: Optional[str]
```
Comment (dmitryduev) [inline wandb/sdk/internal/settings_static.py:55]: add the new policy settings here?

File: wandb/sdk/service/streams.py [hunk 3]
```diff
@@ -321,6 +322,21 @@ def _finish_all(self, streams: Dict[str, StreamRecord], exit_code: int) -> None:
             assert result
             sampled_history = result.response.sampled_history_response
 
+            if (
```
Comment (dmitryduev) [inline wandb/sdk/service/streams.py:325]: Add the finish_timeout/fail logic here, similar to `_on_finish`.

File: wandb/sdk/wandb_init.py [hunk 4]
```diff
@@ -1064,9 +1101,6 @@ def init(
     """
     wandb._assert_is_user_process()
 
-    if resume is True:
```
Comment (dmitryduev) [inline wandb/sdk/wandb_init.py:1067]: Settings take care of this

**Pull request closed without merging**